numpy/core/tests/test_simd.py

```diff
@@ -173,14 +173,21 @@ def test_operators_shift(self):
             # left shift
             shl = self.shl(vdata_a, count)
             assert shl == data_shl_a
-            # left shift by an immediate constant
-            shli = self.shli(vdata_a, count)
-            assert shli == data_shl_a
             # load to cast
             data_shr_a = self.load([a >> count for a in data_a])
             # right shift
             shr = self.shr(vdata_a, count)
             assert shr == data_shr_a
+
+        # shift by zero or max or out-range immediate constant is not applicable and illogical
```

Comment:
shift by zero vector is meaningless too. why not apply this rule to `shl/shr  `?

Reply:
`shl/shr` shift the vector by a constant too. the difference here is `shli/shri` take an "immediate" constant which mean the operand loaded during compile time while the shift operand in `shl/shr` can be a register loaded from memory.
for example
```C
unsigned shift_me = *ptr;
npyv_shri_u32(a, 5); // applicable
npyv_shri_u32(a, shift_me); // not applicable

npyv_shr_u32(a, 5); // applicable
npyv_shr_u32(a, shift_me); // applicable
```